src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1700,6 +1724,20 @@ fn resolution_failure(
                         item(res),
                         variant
                     ),
+                    ResolutionFailure::MalformedGenerics => {
+                        let note = "this link has malformed generics";
+                        let help = "the angle brackets are unbalanced";
+
+                        if let Some(span) = sp {
+                            diag.span_label(span, note);
+                            diag.span_label(span, help);
+                        } else {
+                            diag.note(note);
+                            diag.help(help);
+                        }
+
+                        return;
```

Comment:
This should not `return` here, it's better to use the existing logic. It should also use `span_help` and `span_note` instead of `span_label`. I'm not sure it should use `help` at all though - it's not telling you how to fix the issue, just more details about what went wrong.

Is the 'malformed generics' note that helpful? I feel like it doesn't say that much more than the 'angle brackets' message.

Also, I think the message can be more succinct (https://rustc-dev-guide.rust-lang.org/diagnostics.html#diagnostic-output-style-guide).

```suggestion
                        "unbalanced angle brackets".to_owned()
```
or
```suggestion
                        "mismatched angle brackets".to_owned()
```

Reply:
I changed it to be more succinct 👍 

By the way, the reason I used `span_label` is because `span_help` prints the span again. I'm guessing that's why the [existing logic uses `span_label`](https://github.com/rust-lang/rust/blob/8a28b086e0a42d1b2bcd962f4d99e31b8cd35da9/src/librustdoc/passes/collect_intra_doc_links.rs#L1742-L1746) and not `span_note` as well:

```
warning: unresolved link to `Vec::<Box<T>::new`
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | ... harder? That would be [`Vec::<Box<T>::new()`].
   |                            ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
help: unbalanced angle brackets
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | ... harder? That would be [`Vec::<Box<T>::new()`].
   |                            ^^^^^^^^^^^^^^^^^^^^^
```